### PR TITLE
plots: support dict for x field

### LINKED ISSUE #4120: plots: support for flexible x-axis values
### Report

See https://github.com/iterative/dvc/pull/8421 and https://github.com/iterative/dvc/issues/7754. Need to update plots reference.

### PR DESCRIPTION
**Pull request by dberenbaum**
Closes #4120

### CODE REVIEW

File: content/docs/command-reference/plots/show.md [hunk 1]
```diff
@@ -344,6 +344,55 @@ file:///Users/usr/src/dvc_plots/index.html
 
 ![](/img/plots_show_spec_conf_train_test.svg)
 
+### Sourcing x and y from different files
```
Comment (shcheklein) [inline content/docs/command-reference/plots/show.md:347]: style/minor: x -> `x` 

(inline code)

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 2]
```diff
@@ -625,17 +626,22 @@ Refer to [Visualizing Plots] and `dvc plots show` for examples.
 - `x` (string) - column/field name from which the X axis data comes from. An
   auto-generated _step_ field is used by default.
 
+  - Top-level plots: Accepts string or dictionary. For dictionaries, the keys
```
Comment (shcheklein) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:629]: it's quite hard to unpack this tbh ... do brief snippets right here? Even after reading it a few times, I can't understand all the options :)

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 3]
```diff
@@ -625,17 +626,22 @@ Refer to [Visualizing Plots] and `dvc plots show` for examples.
...
+    are used as paths to data sources. The values (strings) are treated as the
+    source column/field names.
+
+  - Plot outputs: column/field name found in the source plots file.
+
 - `y` - source from which the Y axis data comes from:
```
Comment (shcheklein) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:635]: If it repeats `x` - can we just mention (see `x` for the details)?

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 4]
```diff
@@ -609,7 +609,8 @@ Both individual foreach stages (`train@1`) and groups of foreach stages
 The list of `plots` contains one or more user-defined `dvc plots`
 configurations. Every plot must have a unique ID, which may be either a file or
 directory path (relative to the location of `dvc.yaml`) or an arbitrary string.
-Optional configuration fields can be provided as well.
+If the ID is an arbitrary string, a data source must be provided in the `y`
```
Comment (shcheklein) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:612]: hmm .. is it specifically `y` field?

**Comment by dberenbaum** [comment]
@shcheklein It's pretty confusing to describe accurately, but I tried to reword/reorganize and add an example.

**Review by shcheklein — APPROVED** [review]

**Pull request merged**